src/Nethermind/Nethermind.Consensus.AuRa/Validators/MultiValidator.cs

```diff
@@ -153,7 +156,7 @@ public void PostProcess(Block block, TxReceipt[] receipts, ProcessingOptions opt
                 {
                     if (validator.ValidatorType.CanChangeImmediately())
                     {
-                        SetCurrentValidator(block.Number, validator);
+                        SetCurrentValidator(block.Number, validator, _blockTree.FindParentHeader(block.Header, BlockTreeLookupOptions.RequireCanonical));
```

Comment:
many problems that I had were cause by RequireCanonical searches. Mainly because it is not very thread safe. If you do that in the main blockchain processor then it is ok. Otherwise owrth to discuss.